fuzz/src/msg_targets/utils.rs

```diff
@@ -54,13 +65,14 @@ macro_rules! test_msg_exact {
 			if let Ok(msg) = <$MsgType as Readable<::std::io::Cursor<&[u8]>>>::read(&mut r) {
 				let mut w = VecWriter(Vec::new());
 				msg.write(&mut w).unwrap();
-
 				assert_eq!(&r.into_inner()[..], &w.0[..]);
 			}
 		}
 	}
 }
 
+// Tests a message that must survive roundtrip exactly, modulo one "hole" which may be set to 0s on
+// re-serialization.
```

Comment:
The macros have very different purposes as messages have different expected behavior. You can see which messages have which behavior at fuzz/src/msg_targets/gen_target.sh, but, in general, the *_announcement messages have to round-trip *exactly* (as otherwise we'd invalidate the signatures), most messages just need to round-trip up to the amount of data we know how to interpret, some messages we may throw out invalid stuff (eg if an error message isn't valid UTF-8 we cant String-ize it), so they wont roundtrip correctly. Let me know what I should change to make that more clear.

Reply:
Just pour your Github comment as macros comments :)